packages/ra-core/src/controller/field/ReferenceArrayField.stories.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react';
+import {
+    CoreAdminContext,
+    type GetManyResult,
+    type ListControllerResult,
+    testDataProvider,
+    useReferenceArrayFieldController,
+} from '../..';
+
+const dataProvider = testDataProvider({
+    getMany: (_resource, _params): Promise<GetManyResult> =>
+        Promise.resolve({
+            data: [
+                { id: 1, title: 'bar1' },
+                { id: 2, title: 'bar2' },
+            ],
+        }),
+});
+
+const ReferenceArrayFieldController = props => {
+    const { children, ...rest } = props;
+    const controllerProps = useReferenceArrayFieldController({
+        sort: {
+            field: 'id',
+            order: 'ASC',
+        },
+        ...rest,
+    });
+    return children(controllerProps);
+};
+
+const ReferenceArrayFieldComponent = (props: ListControllerResult) => (
+    <div>
+        <div
+            style={{
+                display: 'flex',
+                alignItems: 'center',
+                gap: '10px',
+            }}
+        >
+            <button
+                onClick={() => props.onSelectAll()}
+                disabled={props.total === props.selectedIds.length}
+            >
+                Select All
+            </button>
+            <button
+                onClick={props.onUnselectItems}
+                disabled={props.selectedIds.length === 0}
+            >
+                Unselect All
+            </button>
+            <p>Selected ids: {JSON.stringify(props.selectedIds)}</p>
+        </div>
+        <ul
+            style={{
+                listStyleType: 'none',
+            }}
+        >
+            {props.data?.map(record => (
+                <li key={record.id}>
+                    <input
+                        type="checkbox"
+                        checked={props.selectedIds.includes(record.id)}
+                        onChange={() => props.onToggleItem(record.id)}
+                        style={{
+                            cursor: 'pointer',
+                            marginRight: '10px',
+                        }}
+                    />
+                    {record.id} - {record.title}
+                </li>
+            ))}
+        </ul>
+    </div>
+);
+
+export const ReferenceArrayField = ({
```

Comment:
I'd call it `Basic` instead

Reply:
I'd do it anyway as this component obviously don't have enough stories and we should/will add more (not in this PR though)